pynetdicom/dimse_messages.py

```diff
@@ -302,6 +307,16 @@ def decode_msg(self, primitive):
                         _MESSAGE_TYPES[self.command_set.CommandField][1]
                     )
 
+                    if (
+                        self.__class__.__name__ == "C_STORE_RQ"
+                        and _config.STORE_RECV_CHUNKED_DATASET
+                    ):
+                        self._data_set_base_dir = TemporaryDirectory()
+                        self._data_set_file = (
+                            Path(self._data_set_base_dir.name)
+                            / f"{uuid.uuid4()}.dcm"
+                        )
```

Comment:
I'd prefer this be a bit lower after the check on `CommandDataSetType == 0x0101`

Reply:
Having the temporary directory lifetime tied to the message lifetime is a good idea